components/list/list-header.js

```diff
@@ -66,14 +75,32 @@ class ListHeader extends RtlMixin(LocalizeCoreElement(LitElement)) {
 			:host([hidden]) {
 				display: none;
 			}
-			.d2l-list-header-container {
+			:host([requires-selection]) .d2l-list-header-container {
+				display: none;
+				max-height: 0;
+				opacity: 0;
+				/* transform: translate(0, -10px); */
+				transition: opacity 100ms linear, max-height 100ms linear;
+			}
+			:host([requires-selection]) .d2l-list-header-container-pre-showing,
```

Comment:
We need the header to be visible in the DOM before animation begins (i.e. we cannot animate starting from `display: none;`.